src/main/java/com/internet/shop/service/ProductServiceImpl.java

```diff
@@ -0,0 +1,42 @@
+package com.internet.shop.service;
+
+import com.internet.shop.dao.ProductDao;
+import com.internet.shop.lib.Inject;
+import com.internet.shop.lib.Service;
+import com.internet.shop.model.Product;
+import java.util.List;
+
+@Service
+public class ProductServiceImpl implements ProductService {
+    @Inject
+    private ProductDao productDao;
+
+    @Override
+    public Product create(Product product) {
+        return productDao.create(product);
+    }
+
+    @Override
+    public Product get(Long id) {
+        if (productDao.get(id).isPresent()) {
+            return productDao.get(id).get();
```

Comment:
you are calling productDao.get() twice and also we can use orElseThrow() with custom exception for this